src/main/java/fi/aalto/cs/intellij/actions/RequiredPluginsCheckerAction.java

```diff
@@ -0,0 +1,211 @@
+
+  private static final Logger logger = LoggerFactory.getLogger(RequiredPluginsCheckerAction.class);
+
+  private static final Map<String, String> requiredPluginNames = new HashMap<>();
+  private static Map<String, String> missingOrDisabledPluginNames = new HashMap<>();
+  private static List<IdeaPluginDescriptor> missingOrDisabledIdeaPluginDescriptors;
+  private static List<IdeaPluginDescriptor> availableIdeaPluginDescriptors;
+
+  /**
+   * The plugin startup activity.
+   */
+  @Override
+  public void runActivity(@NotNull Project project) {
+    populateRequiredPluginNamesMap();
+    filterMissingOrDisabledPluginNames();
+    createListOfMissingOrDisabledPluginDescriptors();
+    checkPluginsStatusAndNotify();
+  }
+
+  /**
+   * Fills the list of required plugin names.
+   *
+   * Later, reading from the from configuration file might occur here.
+   */
+  private void populateRequiredPluginNamesMap() {
+    requiredPluginNames.put("Scala", "org.intellij.scala");
+  }
+
+  /**
+   * Filters out the plugin names that are missing from the current installation.
+   */
+  private void filterMissingOrDisabledPluginNames() {
+    missingOrDisabledPluginNames = requiredPluginNames
+        .entrySet()
+        .stream()
+        .filter(entry -> isPluginMissingOrDisabled(entry.getValue()))
+        .collect(toMap(Entry::getKey, Entry::getValue));
+  }
+
+  /**
+   * Predicate for checking the plugin status.
+   */
+  private static boolean isPluginMissingOrDisabled(String id) {
+    PluginId pluginId = PluginId.getId(id);
+    return !PluginManager.isPluginInstalled(pluginId) || PluginManager
+        .isDisabled(pluginId.getIdString());
+  }
+
+  /**
+   * If there any plugins missing, creates a list of the plugin descriptors for them based on the
+   * publicly available ones.
+   */
+  private void createListOfMissingOrDisabledPluginDescriptors() {
+    if (missingOrDisabledPluginNames.size() > 0) {
+      getAvailablePluginsFromMainRepo();
+      missingOrDisabledIdeaPluginDescriptors = new ArrayList<>();
+      missingOrDisabledPluginNames.forEach((name, id) -> {
+        PluginId missingPluginId = PluginId.getId(id);
+        availableIdeaPluginDescriptors
+            .stream()
+            .filter(
+                availableDescriptor -> availableDescriptor.getPluginId().equals(missingPluginId)
+            )
+            .findFirst()
+            .ifPresent(missingOrDisabledIdeaPluginDescriptors::add);
+      });
+    }
+  }
+
+  /**
+   * Get the list of all the available in the main JetBrains plugin repository.
+   * <p>
+   * Note! The descriptors for the not installed plugins do not exist in IJ until now.
+   */
+  private void getAvailablePluginsFromMainRepo() {
+    try {
+      availableIdeaPluginDescriptors = RepositoryHelper.loadPlugins(new BgProgressIndicator());
+    } catch (IOException ex) {
+      logger.error("Could not retrieve plugins data from main repository.", ex);
```

Comment:
**logging:** a suggested approach to logging. also, do we log any/some events additionally at this point (for say troubleshooting purposes, as currently, the first feedback course stuff gets is when the assignment is submitted) @nikke234))?

Reply:
Juha didn't prioritize that kind of logging, and I'd say we shouldn't implement it because no one has asked us and thus it is unsure if anyone would ever read those logs. (I understand that we are now speaking about logs that would be read by the course staff, not the devs/maintainers of this plugin.)

Reply:
And btw this suggested approach seems very good. :+1:

Reply:
The suggested approach looks good :+1: Additional logging at this point is not needed in my opinion, as it isn't useful right now and it's easy add logging of more events later